src/templates/ScriptTemplateRetriever.ts

```diff
@@ -77,15 +78,42 @@ export class ScriptTemplateRetriever extends TemplateRetriever {
     }
 
     private async parseTemplates(templatesPath: string): Promise<IFunctionTemplate[]> {
-        // only Resources.json has a capital letter
-        this._rawResources = <object>await fse.readJSON(path.join(templatesPath, 'resources', 'Resources.json'));
+        this._rawResources = <object>await fse.readJSON(await getResourcesPath(templatesPath));
         this._rawTemplates = <object[]>await fse.readJSON(path.join(templatesPath, 'templates', 'templates.json'));
         this._rawConfig = <object>await fse.readJSON(path.join(templatesPath, 'bindings', 'bindings.json'));
 
         return parseScriptTemplates(this._rawResources, this._rawTemplates, this._rawConfig);
     }
 }
 
+/**
+ * Unlike templates.json and bindings.json, Resources.json has a capital letter
+ */
+export async function getResourcesPath(templatesPath: string, vscodeLang: string = vscode.env.language): Promise<string> {
+    const folder: string = path.join(templatesPath, 'resources');
+
+    try {
+        // Example: "en-US"
+        const parts: string[] = vscodeLang.split('-');
+        // Example: "en" for "english"
+        const language: string = parts[0];
+        // Example: "US" for "United States" (locale is optional)
+        // tslint:disable-next-line: strict-boolean-expressions
+        const locale: string | undefined = parts[1] || '[a-z]*';
+
+        const files: string[] = await fse.readdir(folder);
+        const regExp: RegExp = new RegExp(`resources\\.${language}(-${locale})?\\.json`, 'i');
+        const matchingFile: string | undefined = files.find(f => regExp.test(f));
```

Comment:
How are the resources done?  Do you always have specific locales, e.g. en-us resources, or can you have just "en" resources?  If the vscode language is "en", and you have en resources and en-uk resources, en should be preferred.

Reply:
This is what the templates look like: https://github.com/Microsoft/vscode-azurefunctions/tree/master/resources/backupScriptTemplates/~2/resources

Not sure if that'll be very common, but fixed